From bdbc51220234b2d937d6b0a8fa60d241c135253f Mon Sep 17 00:00:00 2001
From: Hafizh Pratama <hafizh.pratama@sirclo.com>
Date: Wed, 6 Jul 2022 17:06:09 +0800
Subject: [PATCH] fix([27284331] Icube - Lusitanian): update file abstract

---
 .../Common/Http/Client/AbstractClient.php     | 23 +++++++++----------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/vendor/lusitanian/oauth/src/OAuth/Common/Http/Client/AbstractClient.php b/vendor/lusitanian/oauth/src/OAuth/Common/Http/Client/AbstractClient.php
index 94000b18c..1d1ba1f3e 100644
--- a/vendor/lusitanian/oauth/src/OAuth/Common/Http/Client/AbstractClient.php
+++ b/vendor/lusitanian/oauth/src/OAuth/Common/Http/Client/AbstractClient.php
@@ -3,7 +3,7 @@
 namespace OAuth\Common\Http\Client;
 
 /**
- * Abstract HTTP client
+ * Abstract HTTP client.
  */
 abstract class AbstractClient implements ClientInterface
 {
@@ -23,7 +23,7 @@ abstract class AbstractClient implements ClientInterface
     protected $timeout = 15;
 
     /**
-     * Creates instance
+     * Creates instance.
      *
      * @param string $userAgent The UA string the client will use
      */
@@ -59,15 +59,14 @@ abstract class AbstractClient implements ClientInterface
     /**
      * @param array $headers
      */
-    public function normalizeHeaders(&$headers)
+    public function normalizeHeaders($headers): array
     {
-        // Normalize headers
-        array_walk(
-            $headers,
-            function (&$val, &$key) {
-                $key = ucfirst(strtolower($key));
-                $val = ucfirst(strtolower($key)) . ': ' . $val;
-            }
-        );
+        $normalizeHeaders = [];
+        foreach ($headers as $key => $val) {
+            $val = ucfirst(strtolower($key)) . ': ' . $val;
+            $normalizeHeaders[$key] = $val;
+        }
+
+        return $normalizeHeaders;
     }
-}
+}
\ No newline at end of file
-- 
2.32.1 (Apple Git-133)

