From 9462a39c262f933af501ba43a33cef0ca9dbbff9 Mon Sep 17 00:00:00 2001
From: Hanif Rizal <hanif@icube.us>
Date: Fri, 12 Jun 2020 06:54:52 +0000
Subject: [PATCH] patch for import category image

---
 .../Category/Attribute/Backend/Image.php      | 40 +++----------------
 1 file changed, 5 insertions(+), 35 deletions(-)

diff --git a/vendor/magento/module-catalog/Model/Category/Attribute/Backend/Image.php b/vendor/magento/module-catalog/Model/Category/Attribute/Backend/Image.php
index 4880214e5..13d1effed 100644
--- a/vendor/magento/module-catalog/Model/Category/Attribute/Backend/Image.php
+++ b/vendor/magento/module-catalog/Model/Category/Attribute/Backend/Image.php
@@ -71,8 +71,7 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
 
     /**
      * Gets image name from $value array.
-     *
-     * Will return empty string in a case when $value is not an array.
+     * Will return empty string in a case when $value is not an array
      *
      * @param array $value Attribute value
      * @return string
@@ -87,28 +86,8 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
     }
 
     /**
-     * Check that image name exists in catalog/category directory and return new image name if it already exists.
-     *
-     * @param string $imageName
-     * @return string
-     */
-    private function checkUniqueImageName(string $imageName): string
-    {
-        $imageUploader = $this->getImageUploader();
-        $mediaDirectory = $this->_filesystem->getDirectoryWrite(DirectoryList::MEDIA);
-        $imageAbsolutePath = $mediaDirectory->getAbsolutePath(
-            $imageUploader->getBasePath() . DIRECTORY_SEPARATOR . $imageName
-        );
-
-        $imageName = Uploader::getNewFilename($imageAbsolutePath);
-
-        return $imageName;
-    }
-
-    /**
-     * Avoiding saving potential upload data to DB.
-     *
-     * Will set empty image attribute value if image was not uploaded.
+     * Avoiding saving potential upload data to DB
+     * Will set empty image attribute value if image was not uploaded
      *
      * @param \Magento\Framework\DataObject $object
      * @return $this
@@ -121,15 +100,10 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
 
         if ($this->fileResidesOutsideCategoryDir($value)) {
             // use relative path for image attribute so we know it's outside of category dir when we fetch it
-            // phpcs:ignore Magento2.Functions.DiscouragedFunction
-            $value[0]['url'] = parse_url($value[0]['url'], PHP_URL_PATH);
             $value[0]['name'] = $value[0]['url'];
         }
 
         if ($imageName = $this->getUploadedImageName($value)) {
-            if (!$this->fileResidesOutsideCategoryDir($value)) {
-                $imageName = $this->checkUniqueImageName($imageName);
-            }
             $object->setData($this->additionalData . $attributeName, $value);
             $object->setData($attributeName, $imageName);
         } elseif (!is_string($value)) {
@@ -140,8 +114,6 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
     }
 
     /**
-     * Get Instance of Category Image Uploader.
-     *
      * @return \Magento\Catalog\Model\ImageUploader
      *
      * @deprecated 101.0.0
@@ -182,11 +154,9 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
         $fileUrl = ltrim($value[0]['url'], '/');
         $baseMediaDir = $this->_filesystem->getUri(DirectoryList::MEDIA);
 
-        if (!$baseMediaDir) {
-            return false;
-        }
+        $usingPathRelativeToBase = strpos($fileUrl, $baseMediaDir) === 0;
 
-        return strpos($fileUrl, $baseMediaDir) !== false;
+        return $usingPathRelativeToBase;
     }
 
     /**
-- 
2.20.1

