diff -ruN a/lib/web/mage/validation.js b/lib/web/mage/validation.js
--- a/lib/web/mage/validation.js	2020-10-07 15:51:47.000000000 +0700
+++ b/lib/web/mage/validation.js	2020-10-07 16:08:07.099227200 +0700
@@ -6,10 +6,11 @@
 define([
     'jquery',
     'moment',
+    'mageUtils',
     'jquery-ui-modules/widget',
     'jquery/validate',
     'mage/translate'
-], function ($, moment) {
+], function ($, moment, utils){
     'use strict';
 
     var creditCartTypes, rules, showLabel, originValidateDelegate;
@@ -1032,7 +1033,7 @@
         ],
         'validate-date': [
             function (value, params, additionalParams) {
-                var test = moment(value, additionalParams.dateFormat);
+                var test = moment(value, utils.convertToMomentFormat(additionalParams.dateFormat));
 
                 return $.mage.isEmptyNoTrim(value) || test.isValid();
             },
diff -ruN a/vendor/magento/module-customer/view/frontend/web/js/validation.js b/vendor/magento/module-customer/view/frontend/web/js/validation.js
--- a/vendor/magento/module-customer/view/frontend/web/js/validation.js	2020-10-07 15:51:45.000000000 +0700
+++ b/vendor/magento/module-customer/view/frontend/web/js/validation.js	2020-10-07 16:08:08.385435900 +0700
@@ -1,9 +1,10 @@
 define([
     'jquery',
     'moment',
+    'mageUtils',
     'jquery/validate',
     'mage/translate'
-], function ($, moment) {
+], function ($, moment, utils)  {
     'use strict';
 
     $.validator.addMethod(
@@ -13,7 +14,11 @@
                 return true;
             }
 
-            return moment(value).isBefore(moment());
+            var valueFormatted,
+                inputFormat = 'd/M/yy';
+
+                valueFormatted = moment(value, utils.convertToMomentFormat(inputFormat));
+                return valueFormatted.isBefore(moment());
         },
         $.mage.__('The Date of Birth should not be greater than today.')
     );
